pkgs/tools/archivers/trrntzip/default.nix

```
@@ -0,0 +1,52 @@
+# This is a revival of the old trrntzip at
+# https://sourceforge.net/projects/trrntzip
+#
+# See https://sourceforge.net/p/trrntzip/discussion/457469/thread/d3610ea3b6/
+# there hasn't been any response
+#
+# Besides the new one is on github instead of sourceforge
+# which makes life for us easier
+
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, zlib
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "trrntzip";
+  version = "1.0";
+
+  src = fetchFromGitHub {
+    owner = "0-wiz-0";
+    repo = "trrntzip";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-EqGvP1HbO2PhSoYelhTY0edfwDg2W6LtjoijCoLL5a4=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ zlib ];
+
+  meta = with lib; {
+    description = "The goal of the program is to use standard values when creating zips to create identical files over multiple systems";
+    longDescription = ''
+      Torrentzip converts zip archives to a standard format with some
+      pre-defined values, sorting the files, and using particular compression
+      settings so that running it on zip archives created by other tools will
+      always result in the same output. This helps e.g. with sharing
+      zip archives using BitTorrent (which is where the name comes from).
+
+      This is a revival of https://sourceforge.net/projects/trrntzip.
+    '';
+    homepage = "https://github.com/0-wiz-0/trrntzip";
+    license = with licenses; [
+      # "This software includes code from minizip, which is part of zlib"
+      licenses.zlib
+
+      gpl2Plus
+    ];
+    platforms = platforms.linux;
```

This looks like it should support a whole bunch of platforms (https://github.com/0-wiz-0/trrntzip/blob/main/src/platform.h, https://github.com/0-wiz-0/trrntzip/blob/main/src/platform.c). I don't know how difficult it would be to get it working on darwin, but I think it's better to list out the platforms we think upstream supports, and rather mark darwin as broken.

```suggestion
    platforms = with platforms; unix ++ windows;
```

---

I've always thought that `meta.platforms` is for platforms that were tested, and not for those that may work in theory. https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#built-on-platforms

---

I believe the docs you linked have to do with testing before merge requests. I think the default is to mark supported platforms as either specified upstream, or what we suspect it will run on. I'm on mobile atm, but I'll see if I can find some docs about this later